python/cudf/cudf/core/dataframe.py

```diff
@@ -7297,3 +7304,23 @@ def _get_host_unique(array):
         return [array]
     else:
         return set(array)
+
+
+def _get_columns_from_column_accessor(column_accessor, columns):
```

Comment:
The naming here is confusing. `_get_column_from_column_accessor` implies getting a subset of columns - is that what we are doing?

Reply:
Partially true, it also needs to return empty columns for the non existing columns in `column_accessor`

Reply:
Maybe we just write a function that reindexes a frame in-place:

```python
def _reindex_frame(frame, index, columns, fill_value=None, ...):
     """Reindex a frame in-place"""
```

And use that in `__init__` (as well as reindex)?

Reply:
Made a separate function on similar lines with some cosmetic differences. Please feel to let me know your suggestions